src/Serialization/SparseModules.jl

```diff
@@ -0,0 +1,106 @@
+#=
+# TODO: Serialization of sparse rows makes trouble with the code below. I don't know why.
+@register_serialization_type SRow 
+
+type_params(v::T) where {T<:SRow} = TypeParams(T, base_ring(v))
+
+function save_object(s::SerializerState, v::SRow)
+  save_object(s, v.pos)
+  save_object(s, v.values)
+end
+
+function load_object(s::DeserializerState, ::Type{<:SRow}, parent_ring::Ring)
+  pos = load_object(s, Vector{Int}, 1)
+  vals = load_object(s, Vector{elem_type(parent_ring)}, parent_ring, 2)
+  return sparse_row(parent_ring, pos, vals)
+end
+=#
+
+########################################################################
+# Serialization of `FreeMod`s and their elements
+########################################################################
+
+@register_serialization_type FreeMod uses_id
+
+type_params(F::T) where {T <: FreeMod} = TypeParams(T, :base_ring=>base_ring(F))
```

Comment:
no need to use a pair when it's a single param
```suggestion
type_params(F::T) where {T <: FreeMod} = TypeParams(T, base_ring(F))
```